### PR TITLE
Clean up, categorize and sort unstable features in std.

### PR DESCRIPTION
**Pull request by m-ou-se**

### CODE REVIEW

**Comment by yaahc** [comment]
wait no, it's not passing

@bors r-

File: library/std/src/lib.rs [hunk 1]
```diff
@@ -218,110 +218,117 @@
...
-#![feature(panic_can_unwind)]
-#![feature(panic_unwind)]
-#![feature(platform_intrinsics)]
 #![feature(portable_simd)]
-#![feature(prelude_import)]
 #![feature(ptr_as_uninit)]
 #![feature(raw_os_nonzero)]
-#![feature(rustc_attrs)]
-#![feature(saturating_int_impl)]
-#![feature(slice_internals)]
```
Comment (yaahc) [inline library/std/src/lib.rs:309]: looks like you accidentally deleted this feature in the reorg

File: library/std/src/lib.rs [hunk 2]
```diff
@@ -218,110 +218,117 @@
...
 #![feature(concat_idents)]
 #![cfg_attr(bootstrap, feature(const_fn_fn_ptr_basics))]
 #![cfg_attr(bootstrap, feature(const_fn_trait_bound))]
-#![feature(const_format_args)]
-#![feature(const_io_structs)]
-#![feature(const_ip)]
-#![feature(const_ipv4)]
-#![feature(const_ipv6)]
 #![feature(const_mut_refs)]
-#![feature(const_option)]
```
[the remark this answers is not in the document]
Comment (ChrisDenton) [reply library/std/src/lib.rs:254]: Oh, sorry! I misunderstood that then.

[the remark this answers is not in the document]
Comment (m-ou-se) [reply library/std/src/lib.rs:254]: Oh I did remove some features that I (apparently wrongly) concluded were unnecessary. (Would be nice if we had better tools for this. ^^')

I'll fix it tomorrow. :)

[the remark this answers is not in the document]
Comment (m-ou-se) [reply library/std/src/lib.rs:254]: > tomorrow

:sweat_smile:

**Comment by m-ou-se** [comment]
I've put two missing features back, and rebased. Let's try again.

@bors r=yaahc

**Comment by Dylan-DPC** [comment]
failed in [rollup](https://github.com/rust-lang/rust/pull/95500#issuecomment-1083680972)

@bors r-

**Comment by m-ou-se** [comment]
Turns out `associated_type_bounds` was used in std::sys::sgx. I'll leave it for now and send a separate PR to remove it, since it doesn't really need it.

@bors r=yaahc

**Pull request merged**